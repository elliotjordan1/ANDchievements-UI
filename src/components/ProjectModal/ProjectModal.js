import React from 'react';
import PropTypes from 'prop-types';
import { 
  Background, 
  ModalWrapper, 
  HeadingText, 
  ProjectBody, 
  BodyText, 
  ListItem, 
  TechListItem, 
  StackedList, 
  LogoWrapper, 
  Image, 
  Icon, 
  ListDescription, 
  TechList 
} from './styles';

const ProjectModal = ({ image, onClick, name, client, blurbOne, blurbTwo, blurbThree, ANDis, techStack, logo }) => {
  return (
    <>
    <Background onClick={onClick}>
      <ModalWrapper onClick={onClick}>
        <Image image={image}>
          <LogoWrapper image={logo}/>
        </Image>
          <ProjectBody>
            <HeadingText><em>{client}</em><br />{name}</HeadingText>
            <BodyText>{blurbOne}</BodyText>
            {blurbTwo ? <BodyText>{blurbTwo}</BodyText> : null}
            {blurbThree ? <BodyText>{blurbThree}</BodyText> : null}
            <BodyText><b>ANDis</b> <br /></BodyText>
            <StackedList>
              {ANDis && ANDis.map((item) => {
                  let andi;
                  try {
                    andi = JSON.parse(item);                    
                  } catch {
                    andi = item;
                  }
                   return (
                    <ListItem key={andi.andiId}>
                      <Icon image={andi.imageURL}/>
                      <ListDescription>
                        {andi.name}
                        <em>{andi.andiProjectRole}</em>
                      </ListDescription>
                    </ListItem>);
                  }
                )}
            </StackedList>
            <TechList>
              {techStack && techStack.map((item) => {
                  let tech;
                  try {
                    tech = JSON.parse(item);
                  } catch {
                    tech = item;
                  }
                  return (
                    <TechListItem key={tech.technologyId}>
                      <Icon image={tech.imageURL}/>
                    </TechListItem>);
              }
              )}
            </TechList>
          </ProjectBody>
      </ModalWrapper>
    </Background>
    </>
  );
};

ProjectModal.propTypes = {
  image: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  client: PropTypes.string.isRequired,
  blurbOne: PropTypes.string.isRequired,
  blurbTwo: PropTypes.string,
  blurbThree: PropTypes.string,
  ANDis: PropTypes.arrayOf(PropTypes.object).isRequired,
  techStack: PropTypes.arrayOf(PropTypes.object).isRequired,
  logo: PropTypes.string.isRequired
};

ProjectModal.defaultProps = {
  blurbTwo: null,
  blurbThree: null
};

export default ProjectModal;
