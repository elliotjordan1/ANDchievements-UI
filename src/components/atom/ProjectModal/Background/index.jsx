import styled from 'styled-components';
import theme from '../../../../global/theme';

export default styled.div`
  width: 100vw;
  height 100vh;
  z-index: 2;
  position: fixed;
  background-color: rgba(0, 0, 0, 0.6);
  top: 48px;
  @media ${theme.media.tablet}{
    top: 64px;
  };
  @media ${theme.media.desktop}{
    top: 88px;
  };
`;