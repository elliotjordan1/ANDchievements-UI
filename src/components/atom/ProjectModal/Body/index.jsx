import styled from 'styled-components';
import theme from '../../../../global/theme';

export default styled.div`
line-height: 1.25;
background-color: white;
display: flex;  
width: 100%;
height: 100%;
flex-direction: column;
color: ${theme.colours.brand.grey};
@media ${theme.media.tablet}{
  width: 40%;
  min-width: 280px;
};
@media ${theme.media.desktop}{
  min-width: 530px;
}
`;