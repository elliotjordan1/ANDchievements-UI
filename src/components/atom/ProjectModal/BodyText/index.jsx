import styled from 'styled-components';
import theme from '../../../../global/theme';

export default styled.div`
max-width: 100%;
padding: 16px 20px 0 20px;
font-family: ${theme.fontFamily.primary};
font-size: 12px;
font-weight: 300;
text-align: justify;
max-height: 9em;
@media ${theme.media.tablet}{
  font-size: 16px;
};
@media ${theme.media.desktop}{
  font-size: 24px;
  line-height: 1.5;
}
`;