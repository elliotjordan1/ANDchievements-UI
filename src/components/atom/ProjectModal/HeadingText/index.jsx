import styled from 'styled-components';
import theme from '../../../../global/theme';

export default styled.div`
display: block;
width: 100% - 20px;
text-transform: uppercase;
height: min-content;
font-family: ${theme.fontFamily.primary};
font-weight: 500;
font-family: 'Roboto';
letter-spacing: 0.05em;
padding: 2px 10px;
line-height: 1;
color: ${theme.colours.brand.red};
background-color: ${theme.colours.brand.grey2};
font-size: 28px;
& > em{
  color: ${theme.colours.brand.grey};
  font-size: 20px;
  font-weight: 300;
  font-style: normal;
}
@media ${theme.media.tablet}{
  line-height: 1.1;  
  font-size: 28px;
  padding: 10px 10px;
  & > em{
    color: ${theme.colours.brand.grey};
    font-size: 22px;
    font-weight: 300;
    font-style: normal;
  }
};
@media ${theme.media.desktop}{
  font-size: 38px;
  line-height: 1.1;
  & > em{
    color: ${theme.colours.brand.grey};
    line-height: 0.4;
    font-size: 32px;
    font-weight: 300;
    font-style: normal;
  }
};
`;