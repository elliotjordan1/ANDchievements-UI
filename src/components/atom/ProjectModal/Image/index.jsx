import styled from 'styled-components';
import theme from '../../../../global/theme';

export default styled.div`
background-image: ${ ({ image })  => `url(${image})`};
background-size: cover;
background-position: center;
padding-top: 200px;
width: 100%;
display: flex;
@media ${theme.media.tablet}{
  width: 60%;
  padding: 0;
  height: 100%;
  background-color: white;
  background-size: cover;
  background-repeat: no-repeat;
};
@media ${theme.media.desktop}{
  background-size: cover;
}
`;