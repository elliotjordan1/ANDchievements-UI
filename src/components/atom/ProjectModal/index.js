import Background from './Background';
import Body from './Body';
import BodyText from './BodyText';
import HeadingText from './HeadingText';
import Image from './Image';
import LogoWrapper from './LogoWrapper';
import Wrapper from './Wrapper';

export {
  Background,
  Body,
  BodyText,
  HeadingText,
  Image,
  LogoWrapper,
  Wrapper
}