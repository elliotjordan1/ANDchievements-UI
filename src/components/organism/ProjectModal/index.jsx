/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import { 
  Background, 
  ModalWrapper, 
  HeadingText, 
  ProjectBody, 
  BodyText, 
  StackedList, 
  LogoWrapper, 
  Image
} from './styles';

import { TechListContainer } from '../../atom';
import { ANDiListItem, TechListItem } from '../../molecule';

const ProjectModal = ({project, onClick}) => {
  const { imageurl, name, client, blurb_one, blurb_two, blurb_three, andis, techstack, clientlogourl } = project;

  const andiListComponent = andis && andis.map((item) => {
    const andi = JSON.parse(item);

     return (<ANDiListItem andi={andi} />);
    });

  const techStackComponent = techstack && techstack.map((item) => {
    const tech = JSON.parse(item);

    return (
      <TechListItem tech = {tech} />);
  });

  return (
    <Background onClick={onClick}>
      <ModalWrapper onClick={onClick}>
        <Image image={imageurl}>
          <LogoWrapper image={clientlogourl}/>
        </Image>
          <ProjectBody>
            <HeadingText><em>{client}</em><br />{name}</HeadingText>
            <BodyText>{blurb_one}</BodyText>
            {blurb_two ? <BodyText>{blurb_two}</BodyText> : null}
            {blurb_three ? <BodyText>{blurb_three}</BodyText> : null}
            <BodyText><b>ANDis</b> <br /></BodyText>
            <StackedList>
              {andiListComponent}
            </StackedList>
            <TechListContainer>
              {techStackComponent}
            </TechListContainer>
          </ProjectBody>
      </ModalWrapper>
    </Background>
  );
};

ProjectModal.propTypes = {
  project: PropTypes.objectOf(PropTypes.shape).isRequired,
  onClick: PropTypes.func.isRequired
};

export default ProjectModal;
