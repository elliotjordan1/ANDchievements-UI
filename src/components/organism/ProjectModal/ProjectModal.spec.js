import React from 'react';
import { render } from '@testing-library/react';
import ProjectModal from '.';
import mockData from '../../../assets/mockData.json';

const project = mockData.projects[0];

const mockFunc = jest.fn;

describe('Header component', () => {
  it('renders correctly with mock project', () => {
    const component = <ProjectModal
                        project={project} 
                        onClick={mockFunc}/>;

    const wrapper = render(component);
    
    expect(wrapper).toMatchSnapshot();
  });
});
