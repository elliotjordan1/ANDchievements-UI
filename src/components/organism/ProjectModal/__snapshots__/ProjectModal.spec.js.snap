// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly with mock project 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Background-vxuy5a-0 bnjYss"
      >
        <div
          class="styles__ModalWrapper-vxuy5a-1 ewzIzR"
        >
          <div
            class="styles__Image-vxuy5a-2 jnrJEh"
          >
            <div
              class="styles__LogoWrapper-vxuy5a-3 kvoOQi"
            />
          </div>
          <div
            class="styles__ProjectBody-vxuy5a-5 iFRyfQ"
          >
            <div
              class="styles__HeadingText-vxuy5a-4 jKqfiz"
            >
              <em>
                Big Boss
              </em>
              <br />
              Best Project
            </div>
            <div
              class="styles__BodyText-vxuy5a-6 dGLVIX"
            />
            <div
              class="styles__BodyText-vxuy5a-6 dGLVIX"
            >
              <b>
                ANDis
              </b>
               
              <br />
            </div>
            <div
              class="styles__StackedList-vxuy5a-7 eQleti"
            />
            <ul
              class="styles__TechList-vxuy5a-8 iqmnvP"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Background-vxuy5a-0 bnjYss"
    >
      <div
        class="styles__ModalWrapper-vxuy5a-1 ewzIzR"
      >
        <div
          class="styles__Image-vxuy5a-2 jnrJEh"
        >
          <div
            class="styles__LogoWrapper-vxuy5a-3 kvoOQi"
          />
        </div>
        <div
          class="styles__ProjectBody-vxuy5a-5 iFRyfQ"
        >
          <div
            class="styles__HeadingText-vxuy5a-4 jKqfiz"
          >
            <em>
              Big Boss
            </em>
            <br />
            Best Project
          </div>
          <div
            class="styles__BodyText-vxuy5a-6 dGLVIX"
          />
          <div
            class="styles__BodyText-vxuy5a-6 dGLVIX"
          >
            <b>
              ANDis
            </b>
             
            <br />
          </div>
          <div
            class="styles__StackedList-vxuy5a-7 eQleti"
          />
          <ul
            class="styles__TechList-vxuy5a-8 iqmnvP"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
